lib/aws_command.rb

```diff
@@ -40,6 +49,19 @@ def launch()
         ah.extra_vars['oo_new_inst_tags'].merge!(AwsHelper.generate_host_type_tag(options[:type]))
         ah.extra_vars['oo_new_inst_tags'].merge!(AwsHelper.generate_env_host_type_tag(options[:env], options[:type]))
 
+        # Check if we install a custom openshift
+        if !ENV['OO_OPENSHIFT_BINARY'].nil?
+          ah.extra_vars['oo_openshift_binary'] = ENV['OO_OPENSHIFT_BINARY']
+          FileUtils.cp(ENV['OO_OPENSHIFT_BINARY'], 'roles/openshift_master/files')
+          FileUtils.cp(ENV['OO_OPENSHIFT_BINARY'], 'roles/openshift_minion/files')
+        end
```

Comment:
So the binaries don't have to be under 'files' in order for a role to deploy them. Roles can reference binaries directly off of relative or full paths as well. I don't like this section at all, this should be done completely inside of the role using ansible conditionals and partial or full paths.